[Forwardport] 2.3 develop PR port 15942

eduard13 opened the pull request:
Original Pull Request: https://github.com/magento/magento2/pull/15942

### Fixed Issues (if relevant)
1. magento/magento2#13747: Wysiwyg > Image Uploader >Max width/height

### Manual testing scenarios
Check the task's scenarios

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
[Forward] 2.3 develop PR port 15942

Commit:
Refactored the system configs. Passed imageConfig dependency as optional dependency.

VladimirZaets commented on app/code/Magento/Backend/Block/Media/Uploader.php:
```diff
@@ -44,11 +51,13 @@ class Uploader extends \Magento\Backend\Block\Widget
     public function __construct(
         \Magento\Backend\Block\Template\Context $context,
         \Magento\Framework\File\Size $fileSize,
+        \Magento\Framework\Image\Adapter\ConfigInterface $imageConfig,
```
Hi, @eduard13, due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) we can't add new required parameters to the constructor. 

Please make this param optional, the example you can look [here](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/)

VladimirZaets commented on app/code/Magento/Backend/Block/Media/Uploader.php:
```diff
@@ -30,6 +32,11 @@ class Uploader extends \Magento\Backend\Block\Widget
+    /**
+    * @var \Magento\Framework\Image\Adapter\ConfigInterface
+    */
+    protected $imageConfig;
```
Due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) we can't add protected properties. All new properties should be private.

Commit:
Fixed wrong dependencies

eduard13 commented:
@VladimirZaets thank you for the reviews. I've fixed them, moreover I've added also some other changes, which are fixing some modularity perspectives.
Also I'm not sure what should I do within the failing tests, can you please have a look and advice me something here?
